File: engine/model/src/main/scala/tools/packet/playerinteraction/PlayerShopChat.scala
package tools.packet.playerinteraction

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class PlayerShopChat(private var _name: String, private var _chat: String, private var _slot: Byte) extends PacketInput {
  def name: String = _name

  def chat: String = _chat

  def slot: Byte = _slot

  override def opcode(): SendOpcode = SendOpcode.PLAYER_INTERACTION
}

File: engine/engine-model/src/main/java/tools/packet/character/interaction/PlayerShopChat.java
package tools.packet.character.interaction;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record PlayerShopChat(String name, String chat, Byte slot) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.PLAYER_INTERACTION;
   }
}